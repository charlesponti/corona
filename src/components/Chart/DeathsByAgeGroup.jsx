import React from 'react'
import PropTypes from 'prop-types'
import { Grid, CardContent, Typography, Card } from '@material-ui/core'
import cx from 'classnames'
import styles from './DeathsByAgeGroup.module.css'

import deaths from '../../data/deaths.json'
import { Bar } from 'react-chartjs-2'

export default function DeathsByAgeGroup () {
  return (
    <Grid container justify="center" className={cx(styles.container)}>
      <Grid item component={Card} xs={10}>
        <CardContent>
          <Typography variant="body2">
            <Bar
              data={{
                labels: deaths.map(d => d['Age group']),
                datasets: [
                  {
                    data: deaths.map(d => parseInt(d['COVID-19 Deaths'].replace(',', ''))),
                    label: 'COVID-19 Deaths',
                    borderColor: 'rgba(0, 0, 255, 0.5)',
                    fill: true
                  }
                ]
              }}
            />
          </Typography>
        </CardContent>
      </Grid>
    </Grid>
  )
}

DeathsByAgeGroup.propTypes = {
  countryCode: PropTypes.string
}
