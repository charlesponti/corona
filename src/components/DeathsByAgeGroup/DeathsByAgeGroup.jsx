import React from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Grid from '@material-ui/core/Grid'

import styles from './DeathsByAgeGroup.module.css'
import theme from '../../theme.module.css'

import deathsByAgeGroup from '../../data/deaths.json'
// import { getPercentageOfDeaths } from '../../utils'

export default function DeathsByAgeGroup () {
  return (
    <Grid container justify="center" className={styles.container}>
      <Grid item component={Card} xs={11} className={theme.card}>
        <h2 style={{ paddingLeft: '16px' }}>Deaths by Age Group</h2>
        <CardContent>
          <div className={styles.data}>
            {deathsByAgeGroup.map((row) => {
              const ageGroup = row["Age group"].toLowerCase()
              return (
                <Grid container key={row["Age group"]} className={styles.ageGroup}>
                  <Grid item xs={10} sm={7}>
                    {ageGroup.replace('years', '').replace('year', '')}
                  </Grid>
                  <Grid item xs={2} sm={3} className={styles.deaths}>
                    {row["All Deaths involving COVID-19"]}
                  </Grid>
                  {/* <Grid item xs={2} sm={3}>
                    {getPercentageOfDeaths(row)}%
                  </Grid> */}
                </Grid>
            )})}
          </div>
        </CardContent>
      </Grid>
    </Grid>
  )
}

DeathsByAgeGroup.propTypes = {
  countryCode: PropTypes.string
}
