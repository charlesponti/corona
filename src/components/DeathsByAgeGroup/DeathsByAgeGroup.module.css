.container {
  
}

.ageGroup {
  margin-bottom: 0.25rem;
}

.deaths {
  color: #ff5a5a;
}