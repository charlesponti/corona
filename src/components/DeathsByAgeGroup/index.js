export { default as DeathsByAgeGroup } from './DeathsByAgeGroup'