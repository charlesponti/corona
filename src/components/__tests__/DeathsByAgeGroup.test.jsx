import React from 'react'
import { render } from '@testing-library/react'
import DeathsByAgeGroup from '../Chart/DeathsByAgeGroup'

test('should render ConfirmedCases', async (done) => {
  const { container } = render(
    <DeathsByAgeGroup />
  )
  expect(container).toMatchSnapshot()
  done()
})
