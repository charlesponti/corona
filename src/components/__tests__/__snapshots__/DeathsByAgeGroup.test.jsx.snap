// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmedCases 1`] = `
<div>
  <div
    class="MuiGrid-root container MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          <canvas
            height="150"
            width="300"
          />
        </p>
      </div>
    </div>
  </div>
</div>
`;
